### ISSUE TITLE
A command-line equivalent to exportReferencesGraph?

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nix#1242]
I thought `nix-store` might have an option like that, but `--export` includes the derivation contents and I couldn't find anything else that would fit the pattern. Is there a way, given a store path, to get a file like the output of `exportReferencesGraph`? If I have a Nix expression, I can do something like

```bash
function closure() {
    nix-build --no-out-link -E "with import ./. {}; runCommand \"closure\" { exportReferencesGraph = [ \"x\" ($1) ]; } \"cp x \$out\""
}
```

but if I start from a pre-built store path, I don't know how to query the database to get that information.

cc @edolstra @shlevy

**Comment by copumpkin** [comment]
The closest thing I could find would be `--dump-db`, but that expects to want to dump the entire database rather than just the closure of a particular path; then I'd need to post-process it to extract only the subgraph I want, which sounds tricky.

**Comment by copumpkin** [comment]
For context, I'm trying to unify some logic for copying closures between stores so that I can use it in both a Nix builder (which has easy access to `exportReferencesGraph` but no direct access to the host Nix database) and outside of one, where I can occasionally have pre-built store paths that I need to transport (which I can do with `nix-copy-closure` or `--export` and `--import`, but neither of those would work from inside a builder).

So basically I could fork some of the copying logic and do different things inside a builder and outside it, but it'd be nice to not have to maintain two separate code paths.

**Comment by edolstra** [comment]
`nix path-info -r --json ...` produces the same format as the new (JSON) `exportReferencesGraph`.

**Comment by copumpkin** [comment]
Is there anything:

1. That produces the old format? Not sure what can consume the new JSON format (since I'm consuming it with the old perl script)
2. Available prior to `nixUnstable`?

**Comment by edolstra** [comment]
No and no.

**Comment by copumpkin** [comment]
Oh well, thanks!
